mlflow/models/utils.py

```diff
@@ -487,24 +532,44 @@ def _is_sparse_matrix(x):
         elif isinstance(pf_input, pd.DataFrame):
             new_pf_input = {}
             for col_name, tensor_spec in zip(input_schema.input_names(), input_schema.inputs):
-                new_pf_input[col_name] = _enforce_tensor_spec(
-                    np.array(pf_input[col_name], dtype=tensor_spec.type), tensor_spec
+                pd_series = pf_input[col_name]
+                new_pf_input[col_name] = _reshape_and_cast_pandas_column_values(
```

Comment:
Is this backwards-compatible?

Reply:
I think so.

In mlflow@master, for this case, it does 
```
new_pf_input[col_name] = _enforce_tensor_spec(
  np.array(pf_input[col_name], dtype=tensor_spec.type), tensor_spec
)
```
It only works for scalar type column values.

Now the change it not only works for scalar column values, but also works for list type column values.

Reply:
Awesome!